pytorch_lightning/trainer/trainer.py

```diff
@@ -1150,6 +1150,10 @@ def call_teardown_hook(self, model: LightningModule) -> None:
         self.profiler.teardown(stage=state)
         self.teardown(stage=state)
         model.teardown(stage=state)
+        model._current_fx_name = ""
+        model._current_hook_fx_name = None
+        model._current_dataloader_idx = None
+
```

Comment:
Should these be in a default impl for `model.teardown`?

Reply:
i'd prefer not to since:
- the user isn't aware of these properties
- the user would need to change `teardown` to call `super().teardown()` for this to take effect